grpc/server.go

```diff
@@ -0,0 +1,84 @@
+package sentrygrpc
+
+import (
+	"context"
+
+	"github.com/getsentry/sentry-go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerInterceptor is a grpc interceptor that reports errors and panics
+// to sentry. It also sets *sentry.Hub to context.
+func UnaryServerInterceptor(opts UnaryServerInterceptorOptions) grpc.UnaryServerInterceptor {
+	if opts.ReportOn == nil {
+		opts.ReportOn = ReportAlways
+	}
+
+	return func(
+		ctx context.Context,
+		req interface{},
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (_ interface{}, err error) {
+		hub := sentry.GetHubFromContext(ctx)
+		if hub == nil {
+			hub = sentry.CurrentHub().Clone()
+			ctx = sentry.SetHubOnContext(ctx, hub)
+		}
+
+		defer func() {
+			if r := recover(); r != nil {
+				hub.RecoverWithContext(ctx, r)
+
+				if opts.Repanic {
+					panic(r)
+				}
+
+				err = status.Errorf(codes.Internal, "%s", r)
```

Comment:
Is it intentional that this line will not run if repanic is true?

Reply:
Yes, this is intentional. The idea is to re-panic and to let users do whatever they want. Actually, I copied this re-panic feature from sentryhttp package but maybe we want to remove this feature?

Reply:
Never mind, my mistake. If repanic is true the function will not return and then it doesn't matter if we set `err` or not.